### ISSUE TITLE
Create cache should return back after a small timeout.

### ISSUE THREAD

**Issue by gvsg-rs** [readysettech/readyset#807]
@gvsg-rs said in [REA-3597](https://linear.app/readyset/issue/REA-3597/return-something-from-create-cache#comment-b029be35):

> Lets also do a timeout on the create cache statement, so that it comes back after a certain timeout. We know that the cache would still be created in the background.

**Comment by davisjc** [comment]
Hm, I don't think this is the right direction.  I think the query should block as long as it needs to.  We should make it such that the system is still available via other connections.

**Comment by gvsg-rs** [comment]
If it's a long running command, it should be treated like a job. So create cache will return back with a job id. And someone can look at the job id to see what happened to the command output. Otherwise, if the client disconnects from Readyset, there is no way of knowing what happened to the previous create cache command.

Also I believe the reason we don't want to run multiple create cache commands is because we may not be able to migrate multiple queries concurrently.

**Comment by davisjc** [comment]
I can see the value in nonblocking creations, but I don't think they would be the interface I would prefer as a user: copying a job number and then polling for its completion via a different query.

I think one of the nicest parts of having a SQL interface is being able to fit into a common set of expectations.  If the prompt returns, I expect the work to have completed/errored.  If I hit ^C, I'd expect the work to be aborted.  If my prompt disconnects, I also expect the database to abort the work.  Any divergence from these expectations is a surprise.

There's a lot of precedent for blocking queries in SQL.  The blocking nature enables you to fire off a sequence of queries that each depend on the completion of the previous.  If you are writing a test that wants to use a cache, it becomes 2 lines of SQL if the standard blocking behavior is at play.

If we only support one migration at a time, we can encode that in the SQL interface by blocking, much like competing ALTER commands.

Having said all that, maybe having some kind of asynchronous cache creation would be handy.  I suppose I'd only want the blocking path to also exist as an option.

**Comment by gvsg-rs** [comment]
Fair point. Perhaps we can have two flavors of the command — with and without the async keyword. 'create cache async <SQL>' will return back immediately irrespective of command completion (perhaps returning the query_id), while 'create cache <SQL>' will return only after the command has been executed. Errors from 'create cache async' can be saved or reported in the 'show caches' output.

**Comment by davisjc** [comment]
That sounds pretty nice, actually.  I like the idea of using the query id as an identification of the work and making `SHOW CACHES` show more useful stuff.

**Comment by jasobrown-rs** [comment]
As noted in slack, we already have `create cache concurrently …`, and an accompanying `show readyset migration status {id}`.

**Comment by davisjc** [comment]
Related: [REA-3608](https://linear.app/readyset/issue/REA-3608/caches-made-with-create-cache-concurrently-not-getting-hit)
